src/org/openlcb/cdi/swing/CdiPanel.java

```diff
@@ -1639,7 +1646,31 @@ private String getEventName() {
 
         @Override
         protected void additionalButtons() {
+
             final JTextField tf = textField;
+  
+            JButton bb = factory.handleProduceButton(new JButton("Trigger"));
+            bb.setToolTipText("Click to fire this event.");
+            bb.addActionListener(new java.awt.event.ActionListener() {
+                @Override
+                public void actionPerformed(java.awt.event.ActionEvent e) {
+                    NodeID node = rep.getConnection().getNodeId();
+                    EventID ev = new EventID(org.openlcb.Utilities.bytesFromHexString((String)textField.getText()));
+                    rep.getConnection().getOutputConnection().put(new ProducerConsumerEventReportMessage(node, ev), null);
```

Comment:
let's add node to the second argument of put(...) here.
The second argument is the source of the message. It may be helpful to see that this is a loopback.

Reply:
done